shell/plugins/dashboard-store/mutations.js

```diff
@@ -261,7 +261,12 @@ export default {
     const keyField = getters.keyFieldForType(type);
 
     allLatest.forEach((entry) => {
-      const existing = state.types[type].map.get(entry[keyField]);
+      let existing;
+
+      // need to check if [type] is already a store entry so that we don't have ".map of undefined" errors
+      if (state.types[type]) {
+        existing = state.types[type].map.get(entry[keyField]);
```

Comment:
apparently so... It's been a while since I wrote this code but while testing there was a resource type (I don't which right now, sorry) that had an error exactly here because the type wasn't defined on the store yet.
I know that I left that particular resource list working properly 😅

Reply:
We should chat about this one

Reply:
As an update, this was only happened on `Fleet Clusters` list view afaik. After switching logic to use getters to get the incremental data updates on the list view, error ceased to exist. Couldn't pinpoint the root causes, but it points to some kind of "de-sync" in async operations that caused the `type` not to be set on `list` prop in store when we are checking for existing values for that given `type`.